src/commands/apphosting-secrets-set.ts

```diff
@@ -62,22 +63,27 @@ export const command = new Command("apphosting:secrets:set <secretName>")
       secretValue = fs.readFileSync(dataFile, "utf-8");
     }
 
-    if (!created) {
-      const version = await gcsm.addVersion(projectId, secretName, secretValue);
-      logSuccess(`Created new secret version ${gcsm.toSecretVersionResourceName(version)}`);
-      logSuccess(howToAccess);
-      return;
+    if (created) {
+      logSuccess(`Created new secret projects/${projectId}/secrets/${secretName}`);
     }
 
-    logSuccess(`Created new secret projects/${projectId}/secrets/${secretName}`);
+    const version = await gcsm.addVersion(projectId, secretName, secretValue);
+    logSuccess(`Created new secret version ${gcsm.toSecretVersionResourceName(version)}`);
     logSuccess(howToAccess);
 
+    // If the secret already exists, we want to exit once the new version is added
+    if (!created) {
+      logWarning(grantAccess);
+      return;
+    }
+
     const accounts = await dialogs.selectBackendServiceAccounts(projectNumber, projectId, options);
+
     // If we're not granting permissions, there's no point in adding to YAML either.
     if (!accounts.buildServiceAccounts.length && !accounts.runServiceAccounts.length) {
-      logWarning(
-        `To use this secret your backend, you must grant access. You can do so in the future with ${clc.bold("firebase apphosting:secrets:grantAccess")}`,
-      );
+      logWarning(grantAccess);
+
+      // TODO: For existing secrets, enter the grantSecretAccess dialog only when the necessary permissions don't exist.
```

Comment:
Don't know how we'll ever know this since we don't know what backends this maps code to but 🤷